src/panels/config/devices/ha-config-device-page.ts

```diff
@@ -48,11 +70,45 @@ export class HaConfigDevicePage extends LitElement {
       devices ? devices.find((device) => device.id === deviceId) : undefined
   );
 
+  private _entities = memoizeOne(
+    (
+      deviceId: string,
+      entities: EntityRegistryEntry[]
+    ): EntityRegistryStateEntry[] =>
+      entities
+        .filter((entity) => entity.device_id === deviceId)
+        .map((entity) => {
+          return { ...entity, stateName: this._computeEntityName(entity) };
+        })
+        .sort((ent1, ent2) =>
+          compare(
+            ent1.stateName || `zzz${ent1.entity_id}`,
+            ent2.stateName || `zzz${ent2.entity_id}`
+          )
+        )
+  );
+
   protected firstUpdated(changedProps) {
     super.firstUpdated(changedProps);
     loadDeviceRegistryDetailDialog();
   }
 
+  protected async updated(changedProps): Promise<void> {
```

Comment:
`updated` should not be async. Extract async stuff into a method.

Also, by using `await`, you're calling it in sequence instead of in parallel. If you rewrite it to use `.then` then you can keep it in `updated`